src/Disks/DiskBackup.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include <Disks/IDisk.h>
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <IO/ReadBufferFromFileBase.h>
+#include <IO/WriteBufferFromFile.h>
+#include "Common/Exception.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+class DiskBackup : public IDisk
+{
+private:
+[[noreturn]] void throwNotAllowed() const
+{
+    throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Only read-only operations are supported in DiskBackup");
+}
+
+public:
+    DiskBackup(const String & name_, DiskPtr delegate_, MetadataStoragePtr metadata_);
```

Comment:
As I see this constructor is used only in unit test, because it was not clear how to create a config object from there? If so, take a look here https://github.com/ClickHouse/ClickHouse/blob/e493856ddf4925140c85abd7c205a53caabeff0c/src/Common/tests/gtest_named_collections.cpp#L40-L58 
may be you can do the same, so this constructor could be removed